cpp/include/cudf/column/column_view.hpp

```diff
@@ -367,16 +398,22 @@ class column_view : public detail::column_view_base {
 
   /**
    * @brief Returns the number of child columns.
+   *
+   * @return The number of child columns
    */
   [[nodiscard]] size_type num_children() const noexcept { return _children.size(); }
 
   /**
    * @brief Returns iterator to the beginning of the ordered sequence of child column-views.
+   *
+   * @return An iterator to a `column_view` referencing the first child column
    */
   auto child_begin() const noexcept { return _children.cbegin(); }
 
   /**
    * @brief Returns iterator to the end of the ordered sequence of child column-views.
+   *
+   * @return An iterator to a `column_view` to the element following the last child column
```

Comment:
The use of the word "element" is a little confusing here because we're not talking about elements in a column. I like the language here: https://en.cppreference.com/w/cpp/iterator/end

> an iterator one past the end of the sequence represented by c

```suggestion
   * @return An iterator to a `column_view` one past the end of the child columns
```